src/javascripts/utility/debounce.js

```diff
@@ -0,0 +1,24 @@
+//Source: https://davidwalsh.name/javascript-debounce-function
+
+export function debounce(func, wait, immediate) {
+  let timeout;
+
+  return function executedFunction() {
+    let context = this;
+    let args = arguments;
+	    
+    let later = function() {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+
+    let callNow = immediate && !timeout;
+	
+    clearTimeout(timeout);
+
+    timeout = setTimeout(later, wait);
+	
+    if (callNow) func.apply(context, args);
+  };
+};
+
```

Comment:
prob easier to use `lodash.debounce` rather than adding a new file but nothing wrong with this...maybe even smaller... I dunno